api/v1alpha1/openapi.yaml

```diff
@@ -2549,26 +2521,169 @@ components:
+            The name of the repository resource to use as the sync source for environment variables
+        targetRevision:
+          type: string
+          description: |
+            The specific revision of the repository to pull from
+        envFilePath:
+          type: string
+          description: |
+            Path to a file within the repository that contains environment variables to be consumed.
+      required:
+        - repository
+        - targetRevision
+        - envFilePath
+    KubernetesSecretRepositoryRef:
+      type: object
+      properties:
+        name:
+          type: string
+        namespace:
+          type: string
+        mountPath:
+          type: string
+      required:
+        - name
+        - namespace
+        - mountPath
+    HttpRepositoryRef:
+      type: object
+      properties:
+        repository:
+          type: string
+          description: |
+            The name of the repository resource to use as the sync source
+        suffix:
+          type: string
+          description: |
+            Part of the URL that comes after the base URL. It can include query parameters such as:
+            /path/to/endpoint?query=param
+        filePath:
+          type: string
+          description: |
+            The path of the file where the response is stored in the filesystem of the device.
+      required:
+        - repository
+        - filePath
+    ApplicationSpec:
+      type: object
+      description: "Defines the specification for an application"
+      properties:
+        name:
+          type: string
+          description: "The unique name of the application"
+        bundle:
+          $ref: '#/components/schemas/ApplicationBundleProviderSpec'
+        envVars:
+          $ref: '#/components/schemas/ApplicationEnvVars'
+        envFiles:
+          type: array
+          description: "Environment variable key-value pairs from env files, injected at runtime."
+          items:
+            $ref: '#/components/schemas/GitEnvRepositoryRef'
+      required:
+        - name
+        - bundle
+    ApplicationEnvVars:
+      type: object
+      description: "Environment variable key-value pairs, injected during runtime"
+      additionalProperties:
+        type: string
+    ApplicationBundleImageRef:
+      type: object
+      properties:
+        imageRef:
+          type: string
+          description: "The OCI image reference that defines the bundle"
+      required:
+        - imageRef
+    RenderedApplicationSpec:
+      type: object
+      properties:
+        bundle:
+          type: object
+          properties:
+            image:
+              $ref: '#/components/schemas/ApplicationBundleImageRef'
+            data:
+              $ref: '#/components/schemas/ApplicationBundle'
+          oneOf:
+            - required: ["image"]
+            - required: ["data"]
+        envVars:
+          $ref: '#/components/schemas/ApplicationEnvVars'
+    ApplicationBundle:
```

Comment:
Would the ApplicationBundle not simply the content of the corresponding OCI image / ApplicationBundleImage, i.e. directories and files? Why do we need to create an API for this?

Reply:
Few reasons why
- future proof: in the future bootc will want a secret/configmap like object to apply configs. This object would conform to that pattern.
- metadata: the files themselves don't have metadata such as dependencies and file permissions. If we require an object in the bundle to define this why not have automation to "bundle" this into a standard resource?  That process could also validate the sanity of a bundle.
- file delivery: today we only provide ignition based file delivery we would need a new alternative or use ignition again which feels like overkill

Reply:
going to drop for this iteration of the design. this is a larger problem that requires more thought.